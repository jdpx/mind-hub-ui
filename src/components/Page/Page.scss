.page {
    padding: 66px 0px
}