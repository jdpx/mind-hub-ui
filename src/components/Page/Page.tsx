import React from 'react'
import { Container } from 'react-bootstrap'

import './Page.scss'

interface Props {
    name: string
    children: React.ReactNode
}

export default function Page({ name, children }: Props) {
    return (
        <div className={`page ${name}-page`} data-testid={`${name}-page`}>
            <Container>
                <div className="page-content">{children}</div>
            </Container>
        </div>
    )
}
