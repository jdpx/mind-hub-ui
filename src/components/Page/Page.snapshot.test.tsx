import React from 'react'
import { render } from '@testing-library/react'
import Page from './Page'

describe('Page Snapshots', () => {
    const props = {
        name: 'The page',
    }

    it('renders page with child', () => {
        const { asFragment } = render(
            <Page {...props}>
                <div>Testing</div>
            </Page>,
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
