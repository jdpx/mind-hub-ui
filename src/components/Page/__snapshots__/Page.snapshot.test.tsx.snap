// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Snapshots renders page with child 1`] = `
<DocumentFragment>
  <div
    class="page The page-page"
    data-testid="The page-page"
  >
    <div
      class="container"
    >
      <div
        class="page-content"
      >
        <div>
          Testing
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
